### ISSUE TITLE
TensorFlow/Keras model accumulates system and GPU RAM during training

### ISSUE THREAD

**Issue by CopyOfA** [tensorflow/tensorflow#100085]
### Issue type

Performance

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

binary

### TensorFlow version

2.19.0

### Custom code

Yes

### OS platform and distribution

Red Hat Enterprise Linux 8.10

### Python version

3.10.4

### CUDA/cuDNN version

12.5.20240515

### Current behavior?

I am training a model using TensorFlow/Keras using TensorFlow 2.19.0/Keras 3.10.0. During training, I monitor nvidia-smi and top, and the system RAM and the GPU RAM increase during the training period. Eventually, one or the other runs out of RAM and the training is killed. I don't know why this happens. I am using a custom data generator with inputs as numpy arrays. I am using a custom embedding loss function (triplet semihard loss), that I took directly from the `tensorflow_addons` GitHub page: https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/triplet.py.

### Standalone code to reproduce the issue

```shell
https://colab.research.google.com/drive/11F9T-XgBgeJ9JuAt1DZNfqhm4X5bkenA#scrollTo=VEJLi1Fryv_P


I checked out a T4 GPU.
```

### Relevant log output

```shell

```

**Comment by forest-spells** [comment]
Please try to perform training using some non-TF memory allocator, like jemalloc or TCmalloc, e.g. LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libtcmalloc_minimal.so.4 python3 train.py

**Comment by Venkat6871** [comment]
Hi @CopyOfA ,
Apologies for the delay, and thanks for raising your concern here.
Could you please share the gist properly? The link you provided earlier is not opening. Having the correct gist will be helpful for debugging.

Thank you!

**Comment by CopyOfA** [comment]
I am not sure what you mean by the gist, but I'm guessing you cannot open the Google Colab notebook? Would it be better for me to share some code here in a comment block?

**Comment by Venkat6871** [comment]
Hi @CopyOfA ,
Yes, you are right, the Colab link you provided is not opening for me:
```
https://colab.research.google.com/drive/11F9T-XgBgeJ9JuAt1DZNfqhm4X5bkenA#scrollTo=VEJLi1Fryv_P

```
To help us reproduce and debug, could you please save the notebook as a Colab gist?
In Colab, go to the `File menu` then `Save a copy as GitHub Gist`. This will generate a `.ipynb` file link that I can open directly. 
I am also attaching a screenshot here for your reference.

<img width="764" height="764" alt="Image" src="https://github.com/user-attachments/assets/dfdc1971-51ba-4c95-9664-f4bcf5157dc8" />

Thank you!

**Comment by CopyOfA** [comment]
Does this link work for you: https://colab.research.google.com/gist/CopyOfA/47ddedd99fa017d1410f2b558d09d072/untitled4.ipynb

**Comment by Kayyuri** [comment]
Hi,
Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.
Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
